src/python/fuzzing/fuzzer_selection.py

```diff
@@ -37,20 +37,20 @@ def update_mappings_for_fuzzer(fuzzer, mappings=None):
   entities = ndb_utils.get_all_from_query(query)
   old_mappings = {}
   for entity in entities:
-    old_mappings[(entity.job, entity.platform)] = entity
+    old_mappings[entity.job] = entity
 
   new_mappings = []
   for job_name in mappings:
+    mapping = old_mappings.pop(job_name, None)
+    if mapping:
+      continue
+
     job = data_types.Job.query(data_types.Job.name == job_name).get()
```

Comment:
I don't think we should query everytime, performance benefits are significant here.
also, i dont think job can be associated with multiple platforms, so this optimization works.